src/main.js

```diff
@@ -1,35 +1,71 @@
-    "    \\|__|     \\|__|\\|_______|\\|__|/       \\|__|\\|_______|          \n" +
-    "                                                                   \n" +
-    "                                                                   \n" +
-    "                                                                   \n" +
-    " ________  _______   ___      ___ ___  _______   ___       __      \n" +
-    "|\\   __  \\|\\  ___ \\ |\\  \\    /  /|\\  \\|\\  ___ \\ |\\  \\     |\\  \\    \n" +
-    "\\ \\  \\|\\  \\ \\   __/|\\ \\  \\  /  / | \\  \\ \\   __/|\\ \\  \\    \\ \\  \\   \n" +
-    " \\ \\   _  _\\ \\  \\_|/_\\ \\  \\/  / / \\ \\  \\ \\  \\_|/_\\ \\  \\  __\\ \\  \\  \n" +
-    "  \\ \\  \\\\  \\\\ \\  \\_|\\ \\ \\    / /   \\ \\  \\ \\  \\_|\\ \\ \\  \\|\\__\\_\\  \\ \n" +
-    "   \\ \\__\\\\ _\\\\ \\_______\\ \\__/ /     \\ \\__\\ \\_______\\ \\____________\\\n" +
-    "    \\|__|\\|__|\\|_______|\\|__|/       \\|__|\\|_______|\\|____________|",
-  "color: #d81b60; font-size: 14px; font-weight: bold;"
-);
-
-addEventListener("load", () => {
-  const app = document.querySelector("#app");
-  const buttonImage = document.createElement("img");
-  buttonImage.src = image;
-
-  if (app) {
-    app.appendChild(buttonImage);
+import { updateBanner } from "./components/banner.js";
+import { createMovieList } from "./components/movie-list.js";
+import { createSkeleton, removeSkeleton } from "./components/skeleton.js";
+import { ERROR_API_MESSAGE } from "./constants/movie-contants.js";
+import { getPopularMovies } from "./services/movie-api.js";
+
+addEventListener("load", async () => {
+  const thumbnailList = document.querySelector("main .thumbnail-list");
+  thumbnailList.appendChild(createSkeleton());
+
+  let popularMovieListData;
+  try {
+    popularMovieListData = await getPopularMovies();
+  } catch (error) {
+    removeSkeleton();
+    alert(ERROR_API_MESSAGE);
+    return;
   }
+
+  removeSkeleton();
+  updateBanner(popularMovieListData.results[0]);
+
+  const movieList = popularMovieListData.results;
+  const pager = createPager();
+  const popularMovieList = createMovieList(movieList);
+
+  const moreButton = document.querySelector("main .more");
+  setVisibililty(
+    moreButton,
+    popularMovieListData.total_pages > pager.getPage()
+  );
+  moreButton.addEventListener("click", async () => {
+    moreButton.disabled = true;
+
+    thumbnailList.appendChild(createSkeleton());
+
+    let moreMovieListData;
+    try {
+      moreMovieListData = await getPopularMovies(pager.getNextPage());
+    } catch (error) {
+      removeSkeleton();
+      alert(ERROR_API_MESSAGE);
+      moreButton.disabled = false;
+      return;
+    }
+
+    removeSkeleton();
+    moreButton.disabled = false;
+
+    setVisibililty(moreButton, moreMovieListData.total_pages > pager.getPage());
+
+    const moreMovieList = createMovieList(moreMovieListData.results);
+
+    thumbnailList.appendChild(moreMovieList);
```

Comment:
이번 턴에 말씀드리고 싶었어요!

`let moreMovieListData` 선언이 마음에 걸렸어요. `undefined`로 먼저 선언하고 나중에 할당하는 건 코드 의도가 명확해보이지 않아 보이기 때문이었습니다. 특히, 자바스크립트를 이용하는 환경에서는 타입이 매우 자유롭게 구동되어 추적이 어려우므로 `let`의 사용은 최대한 지양하는 점이 좋을 것 같아요.

기존 코드를 기준으로 getPopularMovies에서 예외가 발생한다면, 바로 catch 블록으로 빠지고 그 작업만 마치고 이하 작업이 실행되지 않습니다. 제 생각에는 그래서 아래와 같이 try 블록에 이하 작업을 넣어주는 편이 보다 명확하다고 생각해요!

```suggestion
    try {
      const moreMovieListData = await getPopularMovies(pager.getNextPage());

      removeSkeleton();
      moreButton.disabled = false;
      setVisibililty(
        moreButton,
        moreMovieListData.total_pages > pager.getPage()
      );

      const moreMovieList = createMovieList(moreMovieListData.results);
      thumbnailList.appendChild(moreMovieList);
    } catch (error) {
      removeSkeleton();
      alert(ERROR_API_MESSAGE);
      moreButton.disabled = false;
      return;
    }
```

Reply:
@sannimdev 
저는 try-catch가 예외가 발생할 수 있는 부분에만 최대한 좁게 사용하는 것을 선호하는데요.
그러다보니 undefined로 먼저 선언이 되고 나중에 할당하는 방식으로 구현을 하게 되었고, 이 것이 결국 자바스크립트와 같은 동적 타입 언어에서 문제가 있다는 것을 인지하지 못했네요. 😅 

아무래도 타입 기반 언어에 익숙하다보니, 적응하는 데 시간이 좀 걸릴 것 같네요. 🙏  
> 특히, 자바스크립트를 이용하는 환경에서는 타입이 매우 자유롭게 구동되어 추적이 어려우므로 let의 사용은 최대한 지양하는 점이 좋을 것 같아요.

호중님 덕분에 이해하였고, try-catch의 범위를 넓혀보았습니다. 🙇 
74b64b824ce4c84834874b7505592fd8742ba8a2